### PR TITLE
consolidation of depth axis arguments / style in plotSPC

### PR DESCRIPTION
**Pull request by dylanbeaudette**
Experimenting with a more modern depth axis for `plotSPC()` along with more options for customization. Changes will include new arguments and removal / deprecation of previous arguments. This could break some old code. Thinking about ways to ease the transition.

  * `plot.depth.axis`
  * `n.depth.ticks`
  * `cex.depth.axis`

## TODO:
- [x] reasonable rounding and interval calculation (#291)
- [x] new argument `depth.axis`
- [x] add / document `interval` argument
- [x] deprecate previous arguments
- [x] document changes
- [x] NEWS
- [x] backwards compatibility
- [x] vignette
- [x] examples
- [x] test a few tutorials
- [x] sketches tutorial
- [x] R CMD check

Reorganized NEWs and added commentary to ["sketches"](https://ncss-tech.github.io/AQP/aqp/sketches.html) tutorial.

Also fixing #290.

### CODE REVIEW

**Comment by dylanbeaudette** [comment]
Quick comparison of "compact" vs. "traditional" axis styles.
image
image

**Comment by dylanbeaudette** [comment]
@brownag I'm going to have to merge this soon. Happy to accept feedback on these topics when you return.

**Comment by dylanbeaudette** [comment]
New comparison, this time with "tape" style. Still generalizing position, always a problem with base graphics.
demo

File: R/sketch-utils.R [hunk 1]
```diff
@@ -1,3 +1,145 @@
...
+  .max <- .roundTo(m, i)
+  
+  # axis sequence
+  .res <- seq(0, .max, by = i)
+  return(.res)
+}
+
+.drawDepthAxis <- function(style = c('compact', 'traditional', 'tape'), .at, .labels, .line, .cex, .n) {
+  
+  style <- match.arg(style)
```
Comment (brownag) [inline R/sketch-utils.R:40]:
```suggestion
  style <- match.arg(tolower(style))
```

File: R/plotMultipleSPC.R [hunk 2]
```diff
@@ -49,27 +49,29 @@
 #'
 #' # plot 2 SoilProfileCollection objects on the same axis
 #' par(mar=c(1,1,1,1))
-#' plot(sp3, n=n.pedons)
-#' plot(sp4, add=TRUE, x.idx.offset=group.ends[1], plot.depth.axis=FALSE, id.style='side')
+#' plot(sp3, n = n.pedons)
```
Comment (brownag) [inline R/plotMultipleSPC.R:52]:
```suggestion
#' plotSPC(sp3, n = n.pedons)
```

File: R/plotMultipleSPC.R [hunk 3]
```diff
@@ -49,27 +49,29 @@
...
+#' plot(sp4, add = TRUE, x.idx.offset = group.ends[1],
```
Comment (brownag) [inline R/plotMultipleSPC.R:53]:
```suggestion
#' plotSPC(sp4, add = TRUE, x.idx.offset = group.ends[1], 
```

**Review by brownag — APPROVED** [review]
Aside from the few minor suggestions below, LGTM!

**Comment by dylanbeaudette** [comment]
Thanks @brownag for those suggestions! I've got a couple of tiny tweaks, and then ready to merge.

**Comment by dylanbeaudette** [comment]
We can do a final vote on the default axis style just before CRAN release.

**Pull request merged**